src/ts/components/settingspanel.ts

```diff
@@ -53,6 +53,8 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
         this.hideHoveredSelectBoxes();
       });
 
+      uimanager.onControlsHide.subscribe(this.hideHoveredSelectBoxes);
```

Comment:
I think the event should be bound outside the if checking for the configuration if the `SettingsPanel` should hide automatically.

Because when the automatic hide for the `SettingsPanel` is disabled the event never get bound and if the whole UI hides the select box is still floating.